src/transactions/feed/TransactionFeedV2.tsx

```diff
@@ -262,6 +279,34 @@ export default function TransactionFeedV2() {
     [isFetching, data?.transactions, originalArgs?.endCursor, standByTransactions.confirmed]
   )
 
+  /**
+   * In order to avoid bloating stand by transactions with confirmed transactions that are already
+   * present in the feed via pagination – we need to cleanup them up. This must run for every page
+   * as standByTransaction selector might include very old transactions. We should use the chance
+   * whenever the user managed to scroll to those old transactions and remove them from persisted
+   * storage. Maybe there is a better way to keep it clean with another saga watcher?
+   */
+  useEffect(
+    function cleanupStandByTransactions() {
+      const confirmedPaginationTransactions =
+        data?.transactions
+          ?.filter((tx) => tx.status !== TransactionStatus.Pending)
+          .map((tx) => tx.transactionHash) || []
+
+      const standByTransactionsToRemove: string[] = []
+      for (const tx of standByTransactions.confirmed) {
+        if (confirmedPaginationTransactions.includes(tx.transactionHash)) {
+          standByTransactionsToRemove.push(tx.transactionHash)
+        }
+      }
+
+      if (standByTransactionsToRemove.length) {
+        dispatch(removeStandByTransactions(standByTransactionsToRemove))
+      }
+    },
+    [data?.transactions, standByTransactions]
```

Comment:
Would it be possible to move this logic to the reducer?
Acting on the successful fetched transactions action.

This way we follow more closely Redux best practices:
- https://redux.js.org/style-guide/#put-as-much-logic-as-possible-in-reducers
- https://redux.js.org/style-guide/#allow-many-reducers-to-respond-to-the-same-action